### PR TITLE
🐛 [amp-story-player] Fix player tests

### PR DESCRIPTION
**Pull request by Enriqe**
Some tests were not running correctly/being flaky.

### CODE REVIEW

File: test/unit/test-amp-story-player.js [hunk 1]
```diff
@@ -37,6 +37,14 @@ describes.realWin('AmpStoryPlayer', {amp: false}, (env) => {
 
   const nextTick = () => new Promise((resolve) => win.setTimeout(resolve, 0));
 
+  function afterRenderPromise() {
+    return new Promise((resolve) => {
+      requestAnimationFrame(() => {
+        setTimeout(resolve);
+      });
+    });
+  }
```
Comment (gmajoulet) [inline test/unit/test-amp-story-player.js:46]: You're combining 3 different methods to wait for the next frame, or schedule a microtask. Can we find something less hacky?
We have some logic in the implementation to remove the poster and show the image on load, can we use something similar for render? That could be useful to publishers as well.

Comment (Enriqe) [reply test/unit/test-amp-story-player.js:46]: Chatted offline. We will replace these checks when the navigation event is ready.

File: test/unit/test-amp-story-player.js [hunk 2]
```diff
@@ -209,43 +217,49 @@ describes.realWin('AmpStoryPlayer', {amp: false}, (env) => {
...
 
-    win.requestAnimationFrame(() => {
-      const iframes = playerEl.shadowRoot.querySelectorAll('iframe');
-      expect(iframes[0].getAttribute('i-amphtml-iframe-position')).to.eql('-1');
-      expect(iframes[1].getAttribute('i-amphtml-iframe-position')).to.eql('0');
-    });
+    const iframes = playerEl.shadowRoot.querySelectorAll('iframe');
+    await afterRenderPromise();
+
+    // TODO(Enriqe): replace with navigation API once ready.
```
Comment (gmajoulet) [inline test/unit/test-amp-story-player.js:228]: Do you want this TODO to be on the `afterRenderPromise`? If you want a ticket plz add its # :))

**Comment by gmajoulet** [comment]
I assume you've seen this, but just in case, you're imported a few extra commits : (

**Comment by Enriqe** [comment]
I think the GH outage caused some random commits tangling here, but I think it's good now.

**Pull request merged**